### PR TITLE
Add PHPStan

### LINKED ISSUE #163: Add static analysis
Add static analysis: 

- [x] PHPStan
- [x] Add to CI

### PR DESCRIPTION
**Pull request by COil**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tickets       | fixes #163 
| License       | MIT
| Doc PR        | NA. Doc in composer.json + phpstan.neon

This adds the PHPStan static analysis at the maximum level.

### CODE REVIEW

File: api/src/Entity/Parchment.php [hunk 1]
```diff
@@ -14,14 +16,12 @@
...
     /**
-     * @var UuidInterface
-     *
      * @ORM\Column(type="uuid", unique=true)
      * @ORM\Id
      * @ORM\GeneratedValue(strategy="CUSTOM")
      * @ORM\CustomIdGenerator(class="Ramsey\Uuid\Doctrine\UuidGenerator")
      */
-    private $id;
+    private UuidInterface $id;
```
Comment (vincentchalamon) [inline api/src/Entity/Parchment.php:24]:
```suggestion
    private ?UuidInterface $id = null;
```

File: api/src/Entity/Parchment.php [hunk 2]
```diff
@@ -34,13 +34,13 @@ public function getId(): ?UuidInterface
      * @Assert\NotBlank
      * @ORM\Column
      */
-    public $title;
+    public string $title;
```
Comment (vincentchalamon) [inline api/src/Entity/Parchment.php:37]:
```suggestion
    public string $title = '';
```
or
```suggestion
    public ?string $title = null;
```

File: api/src/Entity/Parchment.php [hunk 3]
```diff
@@ -34,13 +34,13 @@ public function getId(): ?UuidInterface
...
 
     /**
      * @var string A description of the item
      *
      * @Assert\NotBlank
      * @ORM\Column(type="text")
      */
-    public $description;
+    public string $description;
```
Comment (vincentchalamon) [inline api/src/Entity/Parchment.php:45]:
```suggestion
    public string $description = '';
```
or
```suggestion
    public ?string $description = null;
```

File: api/src/Entity/Review.php [hunk 4]
```diff
@@ -28,46 +30,44 @@
...
     /**
-     * @var UuidInterface
-     *
      * @ORM\Column(type="uuid", unique=true)
      * @ORM\Id
      * @ORM\GeneratedValue(strategy="CUSTOM")
      * @ORM\CustomIdGenerator(class="Ramsey\Uuid\Doctrine\UuidGenerator")
      */
-    private $id;
+    private UuidInterface $id;
```
Comment (vincentchalamon) [inline api/src/Entity/Review.php:38]:
```suggestion
    private ?UuidInterface $id = null;
```

File: api/src/Entity/User.php [hunk 5]
```diff
@@ -14,35 +14,31 @@
...
     /**
-     * @var UuidInterface|null
-     *
      * @ORM\Id
      * @ORM\GeneratedValue(strategy="CUSTOM")
      * @ORM\CustomIdGenerator(class="Ramsey\Uuid\Doctrine\UuidGenerator")
      * @ORM\Column(type="uuid", unique=true)
      */
-    private $id;
+    private ?UuidInterface $id;
```
Comment (vincentchalamon) [inline api/src/Entity/User.php:22]:
```suggestion
    private ?UuidInterface $id = null;
```

File: api/src/Handler/BookHandler.php [hunk 6]
```diff
@@ -73,13 +68,14 @@ public function __invoke(Book $book): void
             return;
         }
 
-        $imageUrl = $contents['data']['memes'][\mt_rand(0, 99)]['url'];
+        $imageUrl = $contents['data']['memes'][\random_int(0, 99)]['url'];
```
Comment (vincentchalamon) [inline api/src/Handler/BookHandler.php:71]: `random_int` renders a cryptographically secure value, but is slower than `mt_rand`. In this case, we don't need a secured value, so I guess it's fine to use `mt_rand` here

**Review by vincentchalamon — CHANGES_REQUESTED** [review]

File: api/phpstan.neon [hunk 7]
```diff
@@ -0,0 +1,11 @@
+includes:
+    - vendor/phpstan/phpstan-symfony/extension.neon
```
[the remark this answers is not in the document]
Comment (COil) [reply api/phpstan.neon:3]: No new alert with these two plugins added.

**Review by vincentchalamon — APPROVED** [review]

**Comment by vincentchalamon** [comment]
Well done @COil Thank you !

**Pull request merged**